Sources/Jetworking/Middleware/Defaults/AuthenticationMiddlewareComponent.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+final class AuthenticationMiddlewareComponent: MiddlewareComponent {
```

Comment:
The idea behind making it final is to prevent overwriting them. The programmer using this framework shall create their own implementation and conform to `MiddlewareComponent` instead of `AuthenticationMiddlewareComponent`. What I could think of is to move the `AuthenticationMethod` enum into this component to be more precise.

What do you think of this?

Reply:
Ok, I think we should keep the method out of the component.

I understand your point with preventing overriding, but why should it be forbidden? For example if we dynamically add header fields, we could just subclass `HeaderFieldsMiddlewareComponent´ and adjust the `headerFields` property. Whats your suggestion to solve the dynamic header fields?

Reply:
The dynamic header fields problem is already solved by setting them when initialising the client configuration (see Testcases). 

In my opinion we should take care about licensing here as well as the warranty which comes with it. The idea behind these components is to have a default working set which can be exchanged by writing own ones. 

Keep in mind when possible problems occur and someone mentions the default component but has adjusted it in some way we are not familiar with and we are searching for an error which is not within this code base we might get stuck or use a lot of time to find out it is not due to our code. 

Summing up these thoughts I would keep it `final` and one then can still copy the contents of these default components and just add their customisations by simply conforming to the `RequestMiddlewareComponent` protocol.

Reply:
The `AuthenticationMethod` enum is now within the `AuthenticationRequestMiddlewareComponent` as it is nowhere else used.

Reply:
Hm ok. But it is not such dynamic as I mean. Sometimes it is necessary to change header while using the same configuration. For example after a login. What do you think about passing a closure which is returning a dictionary of headers? This could be `@autoclosure` to achieve the functionality of your example?

Some thing like in this way:

```swift
import Foundation

public final class HeaderFieldsRequestMiddlewareComponent: RequestMiddlewareComponent {
    private var headerFields: () -> [String: String]

    init(headerFields: @escaping @autoclosure (() -> [String: String])) {
        self.headerFields = headerFields
    }

    public func process(request: URLRequest) -> URLRequest {
        var mutatedRequest: URLRequest = request
        headerFields().forEach { key, value in
            mutatedRequest.addValue(value, forHTTPHeaderField: key)
        }

        return mutatedRequest
    }
}
```

The advantage would be that you can pass a static dictionary or a closure which is executed for each request.

Reply:
Added autoclosure functionality to be highly flexible.